exchanges/okx/okx_wrapper.go

```diff
@@ -1361,75 +1361,67 @@ func (ok *Okx) ValidateCredentials(ctx context.Context, assetType asset.Item) er
-	if err := ok.ValidateKline(pair, a, interval); err != nil {
-		return kline.Item{}, err
-	}
-	if kline.TotalCandlesPerInterval(start, end, interval) > 100 {
-		return kline.Item{}, errors.New(kline.ErrRequestExceedsExchangeLimits)
-	}
-	format, err := ok.GetPairFormat(a, false)
+func (ok *Okx) GetHistoricCandles(ctx context.Context, pair currency.Pair, a asset.Item, interval kline.Interval, start, end time.Time) (*kline.Item, error) {
+	req, err := ok.GetKlineRequest(pair, a, interval, start, end)
 	if err != nil {
-		return kline.Item{}, err
-	}
-	if !pair.IsPopulated() {
-		return kline.Item{}, errIncompleteCurrencyPair
+		return nil, err
 	}
-	instrumentID := format.Format(pair)
-	candles, err := ok.GetCandlesticksHistory(ctx, instrumentID, interval, start, end, 100)
+
+	candles, err := ok.GetCandlesticksHistory(ctx,
+		req.RequestFormatted.Base.String()+
+			currency.DashDelimiter+
+			req.RequestFormatted.Quote.String(),
+		req.ExchangeInterval,
+		start.Add(-time.Nanosecond), // Start time not inclusive of candle.
+		end,
+		300)
 	if err != nil {
-		return kline.Item{}, err
-	}
-	response := kline.Item{
-		Exchange: ok.Name,
-		Pair:     pair,
-		Asset:    a,
-		Interval: interval,
+		return nil, err
 	}
+
+	timeSeries := make([]kline.Candle, len(candles))
 	for x := range candles {
-		response.Candles = append(response.Candles, kline.Candle{
+		timeSeries[x] = kline.Candle{
 			Time:   candles[x].OpenTime,
 			Open:   candles[x].OpenPrice,
 			High:   candles[x].HighestPrice,
 			Low:    candles[x].LowestPrice,
 			Close:  candles[x].ClosePrice,
 			Volume: candles[x].Volume,
-		})
+		}
 	}
-	response.SortCandlesByTimestamp(false)
-	return response, nil
+	return req.ProcessResponse(timeSeries)
 }
 
 // GetHistoricCandlesExtended returns candles between a time period for a set time interval
-func (ok *Okx) GetHistoricCandlesExtended(ctx context.Context, pair currency.Pair, a asset.Item, start, end time.Time, interval kline.Interval) (kline.Item, error) {
-	format, err := ok.GetPairFormat(a, false)
-	if err != nil {
-		return kline.Item{}, err
-	}
-	err = ok.ValidateKline(pair, a, interval)
-	if err != nil {
-		return kline.Item{}, err
-	}
-	instrumentID := format.Format(pair)
+func (ok *Okx) GetHistoricCandlesExtended(ctx context.Context, pair currency.Pair, a asset.Item, interval kline.Interval, start, end time.Time) (*kline.Item, error) {
+	req, err := ok.GetKlineExtendedRequest(pair, a, interval, start, end)
 	if err != nil {
-		return kline.Item{}, err
-	}
-	ret := kline.Item{
-		Exchange: ok.Name,
-		Pair:     pair,
-		Interval: interval,
+		return nil, err
 	}
-	dates, err := kline.CalculateCandleDateRanges(start, end, interval, 100)
-	if err != nil {
-		return kline.Item{}, err
+
+	if count := kline.TotalCandlesPerInterval(start, end, req.ExchangeInterval); count > 1440 {
+		return nil,
+			fmt.Errorf("candles count: %d, exceeds max lookback limit for exchange: %d %w",
```

Comment:
Could you please change this to a different error? It causes confusion when specifically calling the function `ErrRequestExceedsExchangeLimits`

```
 go run . gethistoriccandlesextended --exchange="okx" --pair="btc-usdt" --asset="spot" -i="360" --end="2023-01-18 02:30:00"
2023/01/17 14:01:05 rpc error: code = Unknown desc = candles count: 15129, exceeds max lookback limit for exchange: 1440 request will exceed exchange limits, please reduce start-end time window or use GetHistoricCandlesExtended
exit status 1
```